pkg/scheduler/factory/factory.go

```diff
@@ -489,18 +528,61 @@ func (c *Configurator) GetPredicateMetadataProducer() (predicates.PredicateMetad
 	return getPredicateMetadataProducer()
 }
 
-// GetPredicates returns the predicate functions.
-// It is used by the scheduler and other components, such as k8s.io/autoscaler/cluster-autoscaler.
-func (c *Configurator) GetPredicates(predicateKeys sets.String) (map[string]predicates.FitPredicate, error) {
-	pluginArgs, err := c.getPluginArgs()
+func (c *Configurator) getPredicateConfigs(predicateKeys sets.String) (map[string]predicates.FitPredicate, *config.Plugins, []config.PluginConfig, error) {
```

Comment:
The logic to produce the eventual algorithm vs. framework plugin output has some subtleties. It depends on a few things - algorithm keys in policy, algorithm to plugin config producer registry, ordered list of predicates, etc. I suggest that we add some high level comments here to explain this. Same for the `getPriorityConfig` method.

Reply:
done.